src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Json.cs

```diff
@@ -181,11 +188,11 @@ public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[
 
             if (!ignoreUnsupportedTypes)
             {
-                if (HasUnsupportedType(dataArgType, supportsLazyTypes, out DType unsupportedNestedType, out var unsupportedColumnName))
+                if (HasUnsupportedType(dataArgType, supportsLazyTypes, _settings.MaxDepth, out DType unsupportedNestedType, out var unsupportedColumnName))
```

Comment:
>_settings.MaxDepth [](http://example.com/codeflow?start=71&length=18)

This may break Power Apps - which uses this function _defintion_ directly, and the value of `_settings` will be null here.

Reply:
Good point, let me fix